Python/pywarpx/picmi.py

```diff
@@ -2315,6 +2315,120 @@ def initialize_inputs(self):
             diag.__setattr__('plot_filter_function(t,x,y,z,ux,uy,uz)', expression)
             self.diagnostic._species_dict[name] = diag
 
+
+class BoundaryScrapingDiagnostic(picmistandard.PICMI_ParticleDiagnostic, WarpXDiagnosticBase):
+    """
+    Diagnostic that collects the particles that are absorbed at the boundaries,
+    throughout the simulation.
+
+    Parameters
+    ----------
+    warpx_format: {plotfile, checkpoint, openpmd, ascent, sensei}, optional
+        Diagnostic file format
+
+    warpx_openpmd_backend: {bp, h5, json}, optional
+        Openpmd backend file format
+
+    warpx_file_prefix: string, optional
+        Prefix on the diagnostic file name
+
+    warpx_file_min_digits: integer, optional
+        Minimum number of digits for the time step number in the file name
+
+    warpx_random_fraction: float, optional
+        Random fraction of particles to include in the diagnostic
+
+    warpx_uniform_stride: integer, optional
+        Stride to down select to the particles to include in the diagnostic
+
+    warpx_plot_filter_function: string, optional
+        Analytic expression to down select the particles to in the diagnostic
+    """
+    def init(self, kw):
+
+        self.format = kw.pop('warpx_format', 'plotfile')
```

Comment:
These parts should be in the base class, as they are used across many diagnostics.